### PR TITLE
Add a getting started guide and examples of common tasks

### PR DESCRIPTION
**Pull request by ahawkins**
The getting started is a user level guide. It covers creating a basic assetfile, constructing inputs and matchers, and finally writing a custom filter. 

The examples cover common tasks I've encountered which have confused new users.

I've also updated the readme to point the getting started guide and examples directory. This should help the project feel more documented.

This also closes #91.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -2,3 +2,10 @@
 
 The canonical documentation for Rake::Pipeline is hosted at
 <a href="http://rubydoc.info/github/livingsocial/rake-pipeline/master/file/README.yard">rubydoc.info</a>.
+
+New users are recommened to read `GETTING_STARTED.md` before anything
```
Comment (josemotanet) [inline README.markdown]: Typo: recommened :point_right: recommended

Comment (ahawkins) [inline README.markdown]: Fixed

**Comment by ahawkins** [comment]
@lukemelia anything you think I can add or missed?

**Comment by wagenet** [comment]
@twinturbo I'm certain this is better than nothing. If there are errors, we can fix them later. Unfortunately, I don't have admin access as I thought. @wycats, can you merge this?

**Pull request merged**